src/core/webview/webviewMessageHandler.ts

```diff
@@ -777,26 +762,26 @@ export const webviewMessageHandler = async (provider: ClineProvider, message: We
 					return
 				}
 
-				const existingPrompts = provider.getGlobalState("customSupportPrompts") ?? {}
+				const existingPrompts = getGlobalState("customSupportPrompts") ?? {}
 				const updatedPrompts = { ...existingPrompts }
 				updatedPrompts[message.text] = undefined
-				await provider.updateGlobalState("customSupportPrompts", updatedPrompts)
+				await updateGlobalState("customSupportPrompts", updatedPrompts)
 				await provider.postStateToWebview()
 			} catch (error) {
-				provider.outputChannel.appendLine(
+				provider.log(
 					`Error reset support prompt: ${JSON.stringify(error, Object.getOwnPropertyNames(error), 2)}`,
 				)
 				vscode.window.showErrorMessage(t("common:errors.reset_support_prompt"))
 			}
 			break
 		case "updatePrompt":
 			if (message.promptMode && message.customPrompt !== undefined) {
-				const existingPrompts = provider.getGlobalState("customModePrompts") ?? {}
+				const existingPrompts = getGlobalState("customModePrompts") ?? {}
 				const updatedPrompts = { ...existingPrompts, [message.promptMode]: message.customPrompt }
-				await provider.updateGlobalState("customModePrompts", updatedPrompts)
+				await updateGlobalState("customModePrompts", updatedPrompts)
 				const currentState = await provider.getState()
 				const stateWithPrompts = { ...currentState, customModePrompts: updatedPrompts }
-				provider.view?.webview.postMessage({ type: "state", state: stateWithPrompts })
+				provider.postMessageToWebview({ type: "state", state: stateWithPrompts })
```

Comment:
Use public API, rather than accessing private `view` property.  Had to slightly adjust types on `postMessageToWebview` to enable this - see later.

That might be why the code had been written this way previously.

Reply:
Change of plan - use provider.getStateToPostToWebview() instead, then no type relaxation needed.